### PR TITLE
Issue 30 draft

### PR DESCRIPTION
**Pull request by sahej172002**
not sure how to populate the interface

### CODE REVIEW

File: src/client/Statistics.ts [hunk 1]
```diff
@@ -20,7 +20,50 @@ import {
...
 export function calculateAverages(responseData: GetFullScoresResponseData): SubmissionAverageStatistics {
-  throw new Error("Not Yet Implemented");
+  let cloned = [...responseData.assignments];
+  let subIndices: Array<number | undefined> = [];
+
+  //The for loop iterates through the assignments array
+  //Then it checks if the type is 1 and it is not active
+  //Then if subIndices array doesnt contain the respective submission Index, then add it to subIndices array
+  for (var i = 0; i < cloned.length; i++) {
+    if (cloned[i].type === 1 && !(cloned[i].active)) {
```
Comment (three) [inline src/client/Statistics.ts:30]: You should import and use the `ASSIGNMENT_TYPE_RATING` constant in `constants.ts`.

File: src/client/Statistics.ts [hunk 2]
```diff
@@ -20,7 +20,50 @@ import {
...
+      if ((!subIndices.includes(cloned[i].submissionIndex)) && (!cloned[i].submissionIndex === undefined)) {
```
Comment (three) [inline src/client/Statistics.ts:31]: If the assignment is of type rating, you can assume submissionIndex is valid

File: src/client/Statistics.ts [hunk 3]
```diff
@@ -20,7 +20,50 @@ import {
...
+        sum = 0;
+        count = 0;
+        continue;
+      }
+    }
+    average = sum/count;
+    avgArray.push(subIndices[j],average,0);
+  }
+
+  return avgArray;
```
Comment (three) [inline src/client/Statistics.ts:66]: You should return an array of objects, for instance `[{id: 1, plainAverage: .5}, {id: 2, plainAverage: 3}]` could be a valid return.

Comment (three) [reply src/client/Statistics.ts:66]: You can also remove `judgeWeightedAverage` from the interface. That's lower priority and can be done in a different PR.

File: src/client/Statistics.ts [hunk 4]
```diff
@@ -20,7 +20,50 @@ import {
...
+    var count = 0;
+    var sum = 0;
+    for (var k = 0; k < cloned.length; k++) {
+      if (cloned[k].submissionIndex === subIndices[j]) {
+        if (cloned[k].rating === undefined) {
+          sum = 0;
+          count = 0;
+          continue;
+        }
+        else {
```
Comment (three) [inline src/client/Statistics.ts:51]: Nitpick: keep both brackets of else on the same line,

```javascript
if (/* ... */) {
	// True case
} else {
	// False Case
}
```

File: src/client/Statistics.ts [hunk 5]
```diff
@@ -20,7 +20,50 @@ import {
...
+
+  //Now, run through subInd array and populate result array with scores for each submission from cloned
+  let avgArray: Array<SubmissionAverageStatistics> = [];;
+  for (var j = 0; j < subIndices.length; j++) {
+    var average = 0;
+    var count = 0;
+    var sum = 0;
+    for (var k = 0; k < cloned.length; k++) {
+      if (cloned[k].submissionIndex === subIndices[j]) {
+        if (cloned[k].rating === undefined) {
```
Comment (three) [inline src/client/Statistics.ts:46]: What are you looking for by checking if `rating` is undefined?

File: src/client/Statistics.ts [hunk 6]
```diff
@@ -20,7 +20,50 @@ import {
  * first would be .2*2=.4
  */
 export function calculateAverages(responseData: GetFullScoresResponseData): SubmissionAverageStatistics {
-  throw new Error("Not Yet Implemented");
+  let cloned = [...responseData.assignments];
```
Comment (three) [inline src/client/Statistics.ts:23]: What are you trying to do by cloning `assignments`? If you're trying to make a copy to modify things you need to make a deep copy, but I'm not sure how making a copy helps you.

**Review by three — CHANGES_REQUESTED** [review]

**Comment by three** [comment]
@sahej172002 I added some comments.

You need to return an array matching the interface. For instance you can make an array like:

```js
let stats: SubmissionAverageStatistics = [];
```

and populate with push,

```js
stats.push({ id: 1, plainAverage: 0.5 });
```

But how you make the array is up to you.

**Comment by shivanmodha** [comment]
@sahej172002 what is the status of this one? Can we merge or withdraw?

**Pull request closed without merging**